### PR TITLE
fix sym_storage conversion and some cleanup

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #87731
* __->__ #87718

cc @jansel @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305

### CODE REVIEW

File: torch/_subclasses/meta_utils.py [hunk 1]
```diff
@@ -146,7 +146,7 @@ def meta_tensor(self, t, shape_env=None):
 
         def sym(x):
             if make_symbolic:
-                return shape_env.create_symbol(x)
+                return shape_env.create_symintnode(shape_env.create_symbol(x))
```
Comment (albanD) [inline torch/_subclasses/meta_utils.py:149]: How was this working at all before?

File: test/test_proxy_tensor.py [hunk 2]
```diff
@@ -984,6 +984,14 @@ def f(a, b):
         fx_g = make_fx(f, tracing_mode="symbolic")(torch.randn(16), torch.randn(8))
         self.assertExpectedInline(str(fx_g.shape_env.get_guard_expr()), "Eq(s1, 8) & Eq(s0, 2*s1)")
 
+    def test_sym_storage_offset(self):
+        def f(x, y):
+            return x + y
+
+        inp = (torch.randn(8)[3:], torch.randn(5))
+        fx_g = make_fx(f, tracing_mode="symbolic")(*inp)
+        inp = (torch.randn(8)[3:], torch.randn(5))
```
Comment (albanD) [inline test/test_proxy_tensor.py:993]: Maybe [2:] to make sure we actually trace things properly?

File: torch/_subclasses/meta_utils.py (hunk 1 above)
Comment (Chillee) [reply torch/_subclasses/meta_utils.py:149]: It wasn't :P

Comment (Chillee) [reply torch/_subclasses/meta_utils.py:149]: But I guess we didn't have any tests where we converted an input with a non-zero storageoffset.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**